### ISSUE TITLE
Deletion of lectures fails sometimes

### ISSUE THREAD

**Issue by fosterfarrell9** [MaMpf-HD/mampf#275]
Currently, for a lecture that has assignments or tutorials, the "Delete" but is otherwise empty, the "Delete" button is shown. Pushing it then results in an exception due to foreign key constraint violations.

**Comment by Splines** [comment]
Is this still an issue? If so, could you describe it again or maybe even provide a screen recording?

**Comment by fosterfarrell9** [comment]
Yes it is still an issue (there are some missing `dependent: :destroy` statements in `lecture.rb`).
